app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -99,7 +99,16 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
                 class_name = "middleware_deployment";
                 break;
             case "MiddlewareServer":
-                class_name = "middleware_server";
+                switch (d.item.product) {
+                    case "Hawkular":
+                        class_name = "vendor-hawkular";
```

Comment:
@abonas Right now all 3 types are "sort of WildFly", so they share the same agent code and the differentiator is a product attribute inside the servers management model.
I guess going forward with other middleware like vert.x or Fuse that do not base on WildFly, the separation by model may make more sense.

I think I could change the code in a way that not the product is delivered to the JS-controller, but the icon. That makes it more self-contained and with less places to change. In fact there is already that decorator that I could use - let me change that.